devito/passes/iet/parpragma.py

```diff
@@ -295,15 +295,13 @@ def _select_candidates(self, candidates):
                     except TypeError:
                         pass
 
-                # At least one inner loop (nested) or
-                # we do not collapse most inner loop if it is an atomic reduction
-                if not i.is_ParallelAtomic or nested:
```

Comment:
@FabioLuporini dropped those atomic hacks we added with the original sparse PR. Now just checks which is the outer most offloadable (0 without a device) so we always make sure we don't end up with device pointers outside pragma loops.